### PR TITLE
Transcribe mask ROM uart functions with bedrock2

### PR DESCRIPTION
**Pull request by dayeol**

### CODE REVIEW

**Comment by dayeol** [comment]
Notes: https://hackmd.io/@klQsRT6uTf6T8Ht0igQq-g/BkwtWo5Fd

**Comment by dayeol** [comment]
@googlebot I consent.

File: investigations/bedrock2/Uart/Mmio.v
Comment (jadephilipoom) [inline investigations/bedrock2/Uart/Mmio.v:28]: The MMIO and bitfield files are used in other mask ROM drivers, right? Maybe we should make a separate directory for these shared files such as `investigations/bedrock2/LibBase` (not blocking for this PR though).

File: investigations/bedrock2/Uart/Uart.v
Comment (jadephilipoom) [inline investigations/bedrock2/Uart/Uart.v:29]: Add full C source in a comment here?

File: investigations/bedrock2/Uart/UartToC.v
Comment (jadephilipoom) [inline investigations/bedrock2/Uart/UartToC.v:28]: In order to run the tests, you might need to make a template with wrappers that call the bedrock2-generated functions, like here:

https://github.com/project-oak/silveroak/blob/eb5715d10f5c4ffd2f757ef2400324c08b211751/investigations/bedrock2/Aes/AesToC.v#L199-L202

**Review by jadephilipoom — APPROVED** [review]
Great work!

File: investigations/bedrock2/Uart/Uart.v
Comment (dayeol) [reply investigations/bedrock2/Uart/Uart.v]: resolved

File: investigations/bedrock2/Uart/Mmio.v
Comment (dayeol) [reply investigations/bedrock2/Uart/Mmio.v]: moved both to LibBase

File: investigations/bedrock2/Uart/UartToC.v
Comment (dayeol) [reply investigations/bedrock2/Uart/UartToC.v:34]: I will do this in the next PR if that's ok

**Comment by dayeol** [comment]
@jadephilipoom @satnam6502 I have finished transcribing the functions of maskrom uart, and tested them by manually replacing them in opentitan main repo. I will automate this in the next PR. If that's okay, this PR is ready to go.

**Comment by satnam6502** [comment]
Great! I'll defer to @jadephilipoom and @samuelgruetter for the code review.

**Comment by jadephilipoom** [comment]
> @jadephilipoom @satnam6502 I have finished transcribing the functions of maskrom uart, and tested them by manually replacing them in opentitan main repo. I will automate this in the next PR. If that's okay, this PR is ready to go.

Great! I'll merge in `main` so the CI can run again. I've looked over the changes since last time and my approval stands, so you're all good to merge once the CI passes.

**Pull request merged**
